cli/client_test.go

```diff
@@ -74,3 +83,82 @@ func TestListOrganizationsAction(t *testing.T) {
 	test.That(t, out.messages[1], test.ShouldContainSubstring, "jedi")
 	test.That(t, out.messages[2], test.ShouldContainSubstring, "mandalorians")
 }
+
+func TestTabularDataByFilterAction(t *testing.T) {
+	testMap := map[string]interface{}{"bool": true, "string": "true", "float": float64(1)}
+	pbStruct, err := protoutils.StructToStructPb(testMap)
+	test.That(t, err, test.ShouldBeNil)
+
+	// calls to `TabularDataByFilter` will repeat so long as data continue to be returned,
+	// so we need a way of telling our injected method when data has already been sent so we
+	// can send an empty response
+	dataRequested := false
+	tabularDataByFilterFunc := func(ctx context.Context, in *datapb.TabularDataByFilterRequest, opts ...grpc.CallOption,
+	) (*datapb.TabularDataByFilterResponse, error) {
+		if dataRequested {
+			return &datapb.TabularDataByFilterResponse{}, nil
+		}
+		dataRequested = true
+		return &datapb.TabularDataByFilterResponse{
+			Data:     []*datapb.TabularData{{Data: pbStruct}},
+			Metadata: []*datapb.CaptureMetadata{{LocationId: "loc-id"}},
+		}, nil
+	}
+
+	dsc := &inject.DataServiceClient{
+		TabularDataByFilterFunc: tabularDataByFilterFunc,
+	}
+	resp, err := dsc.TabularDataByFilter(context.Background(), &datapb.TabularDataByFilterRequest{})
+
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, len(resp.Data), test.ShouldEqual, 1)
+
+	protoMap := resp.Data[0].Data.AsMap()
+	test.That(t, protoMap, test.ShouldResemble, testMap)
```

Comment:
Probably not! I suspect we have tests for `StructToStructPB` elsewhere. I included it because it's cheap to do so and the extra sanity check doesn't seem to hurt, but happy to drop it if you think that would be better.

Reply:
I generally like to keep tests short and single purpose so I would